utils/pod_cidr.go

```diff
@@ -41,17 +59,53 @@ func InsertPodCidrInCniSpec(cniConfFilePath string, cidr string) error {
 	if err != nil {
 		return fmt.Errorf("Failed to load CNI conf file: %s", err.Error())
 	}
-	config := make(map[string]interface{})
-	err = json.Unmarshal(file, &config)
-	if err != nil {
-		return fmt.Errorf("Failed to parse JSON from CNI conf file: %s", err.Error())
-	}
+	if strings.HasSuffix(cniConfFilePath, ".conflist") {
+		var config interface{}
+		err = json.Unmarshal(file, &config)
```

Comment:
Wondering if it would help if we unmarshal this into [libcni.NetworkConfig](https://github.com/containernetworking/cni/blob/master/libcni/api.go#L39-L42).

Reply:
or [NetConf](https://github.com/containernetworking/cni/blob/master/pkg/types/types.go#L59-L68)

Reply:
`NetConf` structure is little wierd. 

https://github.com/containernetworking/cni/blob/master/pkg/types/types.go#L70-L72

It does not have the CIDR, did not help much with unmarshalling.